modules/compute/vm-instance/main.tf

```diff
@@ -21,7 +21,19 @@ locals {
   network_storage = var.network_storage != null ? (
   { network_storage = jsonencode(var.network_storage) }) : {}
 
-  resource_prefix = var.name_prefix != null ? var.name_prefix : var.deployment_name
+  # original behavior
+  old_resource_prefix = var.name_prefix != null ? var.name_prefix : var.deployment_name
+
+  # new behavior
+  prefix = var.name_prefix != null ? var.name_prefix : ""
+  deployment = (
+    var.use_deployment_name_in_resource_name == true ? # var could be null
+    var.deployment_name : ""
+  )
+  dash                = local.prefix != "" && local.deployment != "" ? "-" : ""
+  new_resource_prefix = "${local.prefix}${local.dash}${local.deployment}"
+
+  resource_prefix = var.use_deployment_name_in_resource_name == null ? local.old_resource_prefix : local.new_resource_prefix
```

Comment:
I find it confusing that we are comparing a bool to `null` rather than just using the bool. Let's find a way to use it directly.

Reply:
I believe this is resolved. Solution no longer relies on nullness of a boolean.